### ISSUE TITLE
Tearsheet: color layering model doesn't work

### ISSUE THREAD

**Issue by janhassel** [carbon-design-system/ibm-products#3856]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

Carbon's v11 color layering model doesn't work appropriately in the Tearsheet component.

> Is this issue related to a specific component?

Tearsheet

> What did you expect to happen? What happened instead? What would you like to
> see changed?

- readonly inputs should have no visible background / match the background color
  - They have a regular field background in the Tearsheet
- inputs placed in the right influencer are missing their background / match the panel's background
  - even if they're promoted to the next layer, the styles aren't applied

> What browser are you working in?

Safari 17.0

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

`@carbon/ibm-products@2.11.2`

## Steps to reproduce the issue

https://stackblitz.com/edit/github-11rfxa?file=src%2FExample%2FExample.jsx

## Additional information
 
<img width="941" alt="image" src="https://github.com/carbon-design-system/ibm-products/assets/28265588/525ac88a-c2f0-403c-85e9-59e90905d8e3">

**Comment by ljcarot** [comment]
@elycheea Can you investigate if this ever got merged or should be?

**Comment by elycheea** [comment]
@ljcarot Doesn’t look like it so we should move this back into the backlog.

**Comment by ljcarot** [comment]
Since TearSheet is going through overhaul this year, adding @RichKummer for awareness.

**Comment by paul-balchin-ibm** [comment]
Is this a duplicate of #6698? If so, it was fixed and closed.

**Comment by janhassel** [comment]
@paul-balchin-ibm Not quite. Support for the layering model in Tearsheet has improved, but is still not correct.

The right influencer seems to be fine now.
<img width="945" alt="Image" src="https://github.com/user-attachments/assets/6f283eac-cce8-47a3-9faf-d52a7c405dd2" />

However, the left influencer is missing a `<Layer>` wrapper.

Current implementation (2.67.0):
<img width="940" alt="Image" src="https://github.com/user-attachments/assets/1b3a3b77-2057-469e-8b64-57aaf4ef34eb" />

Intended behavior:
<img width="949" alt="Image" src="https://github.com/user-attachments/assets/6311969b-c880-4cc2-9f83-1240ceb576ce" />

Reproduction:
https://stackblitz.com/edit/github-lznkntqf?file=src%2FExample%2FExample.jsx

**Comment by elycheea** [comment]
Following up from this [conversation](https://ibm-studios.slack.com/archives/CQGR0HC05/p1750250223323709?thread_ts=1746011339.318189&cid=CQGR0HC05). We’ll also need to ensure to cover the fixes for the narrow tearsheet that were identified in #7518.

![Image](https://github.com/user-attachments/assets/6e4cfa75-5883-4737-84b7-e228f8da0ec6)
